src/Select/index.js

```diff
@@ -257,7 +257,7 @@ const styles = (theme) => ({
     cursor: (props) => (props.disabled ? "not-allowed" : "default"),
   },
   input: {
-    width: "100%",
+    width: "76%", // fixes weird Firefox bug
```

Comment:
This percentage width scares me– where did this number come from and how does it affect other places where Select is used?

In the case of the demo page– see the attached screenshot for the difference in positioning of the icons for the two different width Select inputs:
<img width="920" alt="screen shot 2018-12-04 at 3 41 41 pm" src="https://user-images.githubusercontent.com/5051745/49477772-24acc100-f7db-11e8-9d03-1d391d21b49f.png">


76% of a 180px box is gonna be different than the 420px box above it– especially when the padding + icon widths are fixed and not percentage based. You can see with the smaller input that the icons are almost getting pushed to the edge/out of the box.

As much as I'm not a huge fan of this– I think the proper way to handle this would be to update the width to `calc(100% - 72px)`, where the 72px is for 12px input padding left/right (24 total), the cancel icon (24), and the dropdown caret icon (24). This will guarantee we never accidentally push the icons outside the box.